internal/provider/resource_workspace_run.go

```diff
@@ -0,0 +1,386 @@
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_source": schema.StringAttribute{
+				MarkdownDescription: "The source of the module, including the API hostname.",
+				Description:         "The source of the module, including the API hostname.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_version": schema.StringAttribute{
+				MarkdownDescription: "The version identifier of the module.",
+				Description:         "The version identifier of the module.",
+				Optional:            true,
+				Computed:            true, // computed if not supplied
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"variables": schema.StringAttribute{
+				MarkdownDescription: "Optional variables for the run in the target workspace.",
+				Description:         "Optional variables for the run in the target workspace.",
+				Optional:            true,
+				// Will remain unset if not supplied.
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+// Configure lets the provider implement the ResourceWithConfigure interface.
+func (t *workspaceRunResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *workspaceRunResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	// Retrieve values from workspace run.
+	var workspaceRun WorkspaceRunModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &workspaceRun)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	created := t.doApplyOrDestroyRun(ctx, workspaceRun, false, resp.Diagnostics)
+
+	// Update the plan with the computed attribute values.
+	t.copyWorkspaceRun(created, &workspaceRun)
+
+	// Set the response state to the fully-populated plan, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, workspaceRun)...)
+}
+
+func (t *workspaceRunResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	// Get the current state.
+	var state WorkspaceRunModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get latest run on the target workspace.
+	toSortBy := sdktypes.RunSortableFieldUpdatedAtDesc // variable needed because can't take address of constant
```

Comment:
The latest run is not guaranteed to equal the latest state since it may have failed or there may have been a manual state import. This will need to call GetCurrentStateVersion on the workspace to get the RunID associated with the latest state.

Reply:
There is no such function in the SDK.  Does it need to be added to the SDK?

Reply:
Thanks for pointing out the function chain that does that.  Fixed.